mission_control/src/mission.cpp

```diff
@@ -36,359 +36,36 @@
-    }
-    m_aborts.clear();
-  }
-
-  current_abort_behavior_id = 0;
-  current_behavior_id = 0;
-}
-
-void Mission::Stop()
-{
-  if (GetState() == MissionState::EXECUTING)
-  {
-    SetState(MissionState::STOPPED);
-
-    // Wait for the mission processing thread to finish
-    ROS_INFO("Stopping mission processing thread");
-    m_threadProcMission->interrupt();  // wake up thread if sleeping
-    m_threadProcMission->join();
-
-    // Clean up all of the behavior publishers and messages
-    ROS_INFO("Stopping and cleaning up active behaviors");
-    Behavior* cur_behavior = getNextBehavior(true);
-    do
-    {
-      cur_behavior->stopBehavior();
-      cur_behavior = getNextBehavior();
-    }
-    while (cur_behavior != NULL);
-  }
-  else if (GetState() == MissionState::ABORTING)
-  {
-    SetState(MissionState::STOPPED);
-
-    ROS_INFO("Stopping abort mission thread");
-    m_threadAbortMission->interrupt();
-    m_threadAbortMission->join();
-
-    // Clean up all of the behavior publishers and messages
-    ROS_INFO("Stopping and cleaning up active behaviors");
-    Behavior* cur_behavior = getNextAbortBehavior(true);
-    do
-    {
-      cur_behavior->stopBehavior();
-      cur_behavior = getNextAbortBehavior();
-    }
-    while (cur_behavior != NULL);
-  }
-}
-
-void Mission::addBehavior(Behavior* behavior) { m_behaviors.push_back(behavior); }
 
-Behavior* Mission::getNextBehavior(bool reset)
-{
-  static bool firsttime = true;
-  if (reset || (firsttime == true))
-  {
-    m_behavior_iterator = m_behaviors.begin();
-    firsttime = false;
-  }
-
-  if (m_behavior_iterator == m_behaviors.end())
-  {
-    return NULL;
-  }
-
-  return *m_behavior_iterator++;
-}
-
-void Mission::addAbortBehavior(Behavior* behavior) { m_aborts.push_back(behavior); }
-
-Behavior* Mission::getNextAbortBehavior(bool reset)
-{
-  static bool abortfirsttime = true;
-  if (reset || (abortfirsttime == true))
-  {
-    m_abort_behavior_iterator = m_aborts.begin();
-    abortfirsttime = false;
-  }
-
-  if (m_abort_behavior_iterator == m_aborts.end())
-  {
-    return NULL;
-  }
-
-  return *m_abort_behavior_iterator++;
-}
-
-void Mission::ProcessCorrectedPoseData(const pose_estimator::CorrectedData& data)
-{
-  if (m_current_behavior == NULL) return;
-
-  if ((GetState() != MissionState::ABORTING) && (GetState() != MissionState::EXECUTING)) return;
-
-  boost::mutex::scoped_lock callback_lock(m_mutCallbacks);
-  if (m_current_behavior->checkCorrectedData(data)) callBackTmr.stop();
-  callback_lock.unlock();
-}
-
-void Mission::SetState(MissionState state)
-{
-  boost::mutex::scoped_lock run_lock(m_MissionStateLock);
-  missionState = state;
-  run_lock.unlock();
-}
+using mission_control::Mission;
+using namespace mission_control;
+using namespace BT;
 
-Mission::MissionState Mission::GetState()
+Mission::Mission(BT::Tree&& missionTree) : tree_(std::move(missionTree))
 {
-  MissionState temp;
+    description_ = tree_.rootNode()->name();
```

Comment:
@patapufate meta: hmm, this seems insufficient. OK to move forward as-is, but we need better, more verbose descriptions in the future (a separate field in the `Mission` message? using XML comments?)